eth/catalyst/api.go

```diff
@@ -516,22 +516,24 @@ func (api *ConsensusAPI) newPayload(params engine.ExecutableData) (engine.Payloa
 		err   error
 	)
 	if api.eth.BlockChain().Config().Taiko && params.Transactions == nil {
+		h := calcWithdrawalsRootTaiko(params.Withdrawals)
```

Comment:
Not sure if we should also pass the new calculated root hash here: https://github.com/taikoxyz/taiko-geth/blob/taiko/eth/catalyst/api.go#L353

Reply:
OK, I alsoi think it needed to be added to `types.NewBlockWithHeader` so i did that as well, then called via `FinalizeBlockAndAssemble` in taiko consensus